apps/req.c

```diff
@@ -644,25 +643,17 @@ int req_main(int argc, char **argv)
                 genopt = sk_OPENSSL_STRING_value(pkeyopts, i);
                 if (pkey_ctrl_string(genctx, genopt) <= 0) {
                     BIO_printf(bio_err, "parameter error \"%s\"\n", genopt);
-                    ERR_print_errors(bio_err);
                     goto end;
                 }
             }
         }
 
-        if (pkey_type == EVP_PKEY_EC) {
-            BIO_printf(bio_err, "Generating an EC private key\n");
-        } else {
-            BIO_printf(bio_err, "Generating a %s private key\n", keyalgstr);
-        }
-
         EVP_PKEY_CTX_set_cb(genctx, genpkey_cb);
         EVP_PKEY_CTX_set_app_data(genctx, bio_err);
 
-        if (EVP_PKEY_keygen(genctx, &pkey) <= 0) {
-            BIO_puts(bio_err, "Error Generating Key\n");
+        if (!app_keygen(genctx, &pkey,
+                        pkey_type == EVP_PKEY_EC ? "EC": keyalgstr, newkey_len))
```

Comment:
I note that the purpose of 57358a83a401ef469353b7ebdae0cf3b870a4d5e was to get the bits right, not something specific about EC as the algorithm name.

Reply:
Meanwhile the `newkey_len` appears to be set correctly also in the EC case.

Reply:
Ah yes, and the special case for the key alg name is not (any more) needed either, so removed that.